src/server/data-access/rosDescriptor/index.ts

```diff
@@ -17,17 +18,48 @@ const getExecutiveReport = async (headers: AxiosRequestHeaders) => {
   return { data };
 };
 
-const getMock: ServiceCallFunction = () => Promise.resolve({ data: rosData });
-const responseProcessor = (data: typeof rosData) => data;
+const getSystemsReport = async (
+  headers: AxiosRequestHeaders,
+  { params }: Record<string, any>
+) => {
+  const defaultParams = {
+    order_by: 'report_date',
+    order_how: 'desc',
+    limit: -1,
+    ...params,
+  };
```

Comment:
@Hyperkid123 while making call to `/systems` api it can happen that we would like to pass query params for filter options and hence I have added this code assuming that we can pass the same query params to `/crc-pdf-generator/v1/generate` api. I hope thats correct.

Also I was going through the code to get access to the `filter query params` as we show them in the systems PDF but couldn't find a way to return the query params data in `RosSystemsTemplate` method. Can you guide a bit here please on achieving the same? Thanks 🙌🏻

Reply:
here is screenshot of how we consume selected filter info in the PDF for your reference
![image](https://user-images.githubusercontent.com/5928530/231111182-bf2e48e2-3d7a-489a-ae7a-84758e53b0d9.png)

Reply:
@PreetiW you can use the post body in your browser request to add additional arguments to your template/api descriptor. They will be accessible as a second argument in your descriptor.

The only way to pass any data to the template is from the descriptor. So I would say get the data from the descriptor and then pass it to the template. You can create any props object you like.